lib/primitives/mtx.js

```diff
@@ -100,15 +100,21 @@ class MTX extends TX {
   }
 
   /**
-   * Clone the transaction. Note that
-   * this will not carry over the view.
+   * Clone the transaction and it's input coins.
    * @returns {MTX}
    */
 
   inject(mtx) {
     assert(mtx instanceof this.constructor);
     super.inject(mtx);
     this.changeIndex = mtx.changeIndex;
+
+    // CoinView has other properties like UndoCoins and BitView
+    // that an MTX doesn't care about. We just want a copy of the
+    // coins spent by the MTX.
+    for (const [key, value] of mtx.view.map.entries())
```

Comment:
The reason behind not cloning the view is that in most cases it's not necessary and it's complex.
Also we have couple of variations on the view. Notably, the new `WalletCoinView` needs different approach to be properly cloned.

So in most cases `mtx` clone is done like:
```
const nmtx = mtx.clone();
nmtx.view = mtx.view;
```

Referencing the old view. It needs double checking if there's a case where old way messes things up, but I doubt. Most views in the wallet side are short lived.